conans/client/graph/graph_binaries.py

```diff
@@ -399,6 +399,17 @@ def _compute_package_id(self, node, default_package_id_mode, default_python_requ
         if apple_clang_compatible:
             conanfile.compatible_packages.append(apple_clang_compatible)
 
+        # opt-in compiler compatibility
+        try:
+            compiler_name = conanfile.settings.compiler.value
+            compiler_version = conanfile.settings.compiler.version.value
+            for compat_version in getattr(profile_host.processed_settings.compiler_compatibility[compiler_name], compiler_version).values_range:
```

Comment:
What is this ``compiler_compatibility``?

Reply:
This is a dictionary in the settings.yml where the user can add compatible compilers ... (perhaps `compatible_compilers` would be indeed a better name for it).